rosbag2_storage/include/rosbag2_storage/storage_interfaces/base_info_interface.hpp

```diff
@@ -30,6 +32,13 @@ class ROSBAG2_STORAGE_PUBLIC BaseInfoInterface
   virtual ~BaseInfoInterface() = default;
 
   virtual BagMetadata get_metadata() = 0;
+
+  /**
+   * Retrieves the relative path to the backing of the storage plugin.
+   *
+   * \returns the relative path.
+   */
+  virtual std::string get_relative_path() const = 0;
```

Comment:
I don't think this function should be on the `BaseInfoInterface` class. This would somehow conflict with `get_metadata()` as I would expect the `get_metadata()` to have a fleshed out metadata with all relative paths set (in the reader case for example).

Reply:
PR #184 adds `generate_metadata` to `Writer` and PR #185 replaces the calls to the storage plugin's `get_metadata` with `Writer`'s `generate_metadata`.

Reply:
Can we enhance this PR then to be somewhat feature complete? I think that makes reviewing the PR easier.

Reply:
I can merge this PR with #184 in that case. I was trying to keep the number of changes to a minimum.

Reply:
that would make sense to me. I appreciate splitting up changes into multiple PRs, however I'd like to have them clustered by features. So one feature could be to extract the metadata generation from the storage plugins into a writer/reader API.
In this particular case, please also make sure that this change has to be applicable to the `rosbag2_bag_v2` plugin, which might not have a `metadata.yaml` file: https://github.com/ros2/rosbag2_bag_v2

Reply:
Ok. I'll close this PR after I finish integrating these changes into #184